runtime/plaid/src/benchmark/mod.rs

```diff
@@ -0,0 +1,184 @@
+use serde::Deserialize;
+use std::collections::HashMap;
+use tokio::fs::File;
+use tokio::io::AsyncWriteExt;
+use tokio::sync::mpsc::Receiver;
+use tokio::time::{timeout, Duration};
+use tokio_util::sync::CancellationToken;
+
+#[derive(Deserialize)]
+pub struct Benchmarking {
+    /// The full path to the output file where benchmarking metrics should be written
+    #[serde(default = "default_results_file_path")]
+    output_file_path: String,
+}
+
+/// Default file path for benchmarking results if none is provided in the config
+fn default_results_file_path() -> String {
+    format!(
+        "{}/../benchmark-results/metrics.txt",
+        env!("CARGO_MANIFEST_DIR")
```

Comment:
I think this is an absolute path which we don't want to pull into compilation as the binary could be moved anywhere later. I would just keep it relative in the same folder or in an adjacent subfolder.

Reply:
Yeah good call